### PR TITLE
First draft of sensorpush summary plot to first page

### PR DESCRIPTION
**Pull request by sofiahaglund**
I will try to add a loading spinner and change the color of the bands. 

Phil has some idea on how to format the data points to speed up the loading!

### CODE REVIEW

File: run_dir/static/js/index.js [hunk 1]
```diff
@@ -118,7 +118,103 @@ $('body').on('click', '.group', function(event) {
...
+    var frig_series = [];
+    var freez_series = [];
+    $.each(data, function(id, sensordata){
+      var timedata = sensordata.samples;
+      var sensname = sensordata.sensor_name;
+      var sercolor = '';
+      for (i in timedata) {
+        timedata[i][0] = Date.parse(timedata[i][0]);
+        date = new Date()
+        date.setTime(timedata[i][0])
```
Comment (ewels) [inline run_dir/static/js/index.js:132]: I can't see where this variable is used?

File: run_dir/static/js/index.js [hunk 2]
```diff
@@ -118,7 +118,103 @@ $('body').on('click', '.group', function(event) {
...
 });
 
+function plot_sum_data(){
+  $.getJSON("/api/v1/sensorpush", function(data) {
+    var frig_series = [];
+    var freez_series = [];
+    $.each(data, function(id, sensordata){
+      var timedata = sensordata.samples;
+      var sensname = sensordata.sensor_name;
+      var sercolor = '';
```
Comment (ewels) [inline run_dir/static/js/index.js:128]: This doesn't seem to be used, can be removed?

**Review by ewels — COMMENTED** [review]
Suggestion is to plot either a rolling-average or a subsampled-average for each data point. This is mostly to smooth out the lines to make it visually cleaner. If done early in the processing (Python, or even better CouchDB) then it could also speed up plotting, if subsampling. Maybe (maybe not).

Looking at this PR I can see that this would need to be done back in the code by @alneberg though really, as none of the new code here is working with the data series themselves.

File: run_dir/static/js/index.js (hunk 2 above)
Comment (sofiahaglund) [reply run_dir/static/js/index.js:128]: True! It's from something I tried before :-)

File: run_dir/static/js/index.js (hunk 1 above)
Comment (sofiahaglund) [reply run_dir/static/js/index.js:132]: I think that's just how you set the time in JS? 
I copied the logic from here: https://github.com/SciLifeLab/genomics-status/blob/master/run_dir/static/js/sensorpush.js#L3

But I could be wrong 🤔

**Comment by alneberg** [comment]
@ewels, yes I agree smoothing out the curves could probably speed it up if done early, but I think the major issue is my choice to split up the data on 1 document per day and sensor in Couchdb, so one needs to fetch quite a few documents in order to get all the data needed. Maybe one can write a clever view in couchdb for that though

File: run_dir/static/js/index.js [hunk 3]
```diff
@@ -118,7 +118,104 @@ $('body').on('click', '.group', function(event) {
...
+        frig_series.push(dp_frig);
+      }
+      else if (sensname.startsWith('F') || sensname == 'TestF35'){
+        var dp_freez = {
+          name: sensname,
+          data: timedata,
+          lineWidth: 1
+        };
+        freez_series.push(dp_freez);
+      }
```
Comment (aanil) [inline run_dir/static/js/index.js:146]:
```suggestion
      var dp_var = {
          name: sensname,
          data: timedata,
          lineWidth: 1
        };
      if (sensname.startsWith('K') || sensname == 'Test F36'){
        frig_series.push(dp_var);
      }
      else if (sensname.startsWith('F') || sensname == 'TestF35'){
        freez_series.push(dp_freez);
      }
```

File: run_dir/static/js/index.js (hunk 1 above)
Comment (ewels) [reply run_dir/static/js/index.js]: It is yeah, but you are setting it in a variable called `date` which is never used.

Comment (sofiahaglund) [reply run_dir/static/js/index.js]: You're right!

File: run_dir/static/js/index.js (hunk 3 above)
Comment (sofiahaglund) [reply run_dir/static/js/index.js]: Nice!

**Comment by sofiahaglund** [comment]
With the indicators-module it's possible to create a sma-line: https://api.highcharts.com/highstock/plotOptions.sma?_ga=2.154446456.1162058945.1643716057-555367210.1641907535

Can't really get it to work though 🤔

**Comment by ewels** [comment]
Nice, though that won't reduce the load on the database so not sure if it will help speed up the load time at all. I suspect that modifying the underlying data query will be easier..

**Review by ewels — APPROVED** [review]
Looking great! Good to merge now I think 🚀

**Review by aanil — APPROVED** [review]
👍

**Pull request merged**